### PR TITLE
Show Anki card flags

### LINKED ISSUE #333: flags and editing QoL
When you mine a card and its in Anki theres a little book icon to show you mined it which you can click on to open the card, when I mine a word that doesnt have audio I like to mark them with a flag in anki and potentially change them when i find it being said later down the line to improve the card....

**_First suggestion:_** If a card is flagged in anki, can yomitan show a flag next to the book so that you dont need to manually open the card to see if its flagged or not
**_Second suggestion:_** Is there a way to trick yomi/windows/sharex to be able to use the same macros that add picture/audio on creation to be able to update/edit the card, for example: if i found a flagged card i wanted to edit, could i push a button on yomitan to overwrite the original cards sentence adding the new one AND then be able to use the same picture/audio capture on sharex to add/overwrite the picture/audio that was originally on there, that way other binds/manually copy pasting etc can be streamlined

its not so bad for cards I havent repped as then its fine to delete the card and recreate, but if its already been repped then it needs manual editting which just takes time away from immersion.

### PR DESCRIPTION
**Pull request by Kuuuube**
Fixes #333

~~Blocking on AnkiConnect lacking support for flags. I've contacted foosoft to get support for this merged in. Currently using this patch to test: [0001-Add-support-for-flags.txt](https://github.com/user-attachments/files/17637823/0001-Add-support-for-flags.txt).~~

Older AnkiConnect versions without flag support will still work fine, there are no errors, it just wont be able to show any flags.

### CODE REVIEW

**Comment by Kuuuube** [comment]
This is all good to go now, just blocking on AnkiConnect support.

Single flag/multiple flags of the same color:
image

Multiple flags of different colors:
image

**Comment by jamesmaa** [comment]
Does this PR handle if a user has an older version of ankiconnect that hasn't supported tags yet?

**Comment by Kuuuube** [comment]
Support for flags has been merged into AnkiConnect. This PR is no longer blocked.

File: ext/js/display/display-anki.js [hunk 1]
```diff
@@ -508,6 +535,104 @@ export class DisplayAnki {
...
+            Purple: {red: 192, green: 132, blue: 252},
+        };
+
+        const gradientSliceSize = 100 / flags.size;
+        let currentGradientPercent = 0;
+
+        let gradient = 'linear-gradient(to right,';
+        for (const flag of flags) {
+            const flagColor = flagColorsDict[flag];
+            gradient += 'rgb(' + flagColor.red + ',' + flagColor.green + ',' + flagColor.blue + ') ' + currentGradientPercent + '%,';
```
Comment (jamesmaa) [inline ext/js/display/display-anki.js:614]: Wow is there really no better way to construct this than string interpolation?

File: ext/settings.html [hunk 2]
```diff
@@ -1960,7 +1960,7 @@ <h1>Yomitan Settings</h1>
             <div class="settings-item-inner">
                 <div class="settings-item-left">
                     <div class="settings-item-label">
-                        Show card tags
+                        Show card tags and flags
```
Comment (jamesmaa) [inline ext/settings.html:1963]: Would it be annoying to rename the setting to `anki.displayTagsAndFlags`?

**Review by jamesmaa — DISMISSED** [review]

File: ext/js/display/display-anki.js (hunk 1 above)
Comment (Kuuuube) [reply ext/js/display/display-anki.js:614]: Not really. I could make it an array of strings then join them to remove the silly "comma remover" slice I guess. Making CSS in js is messy especially for something complex like this which is a 2-14 part gradient.

File: ext/settings.html (hunk 2 above)
Comment (Kuuuube) [reply ext/settings.html:1963]: Wouldnt be too bad but idk if it's worth bothering.

File: ext/js/display/display-anki.js (hunk 1 above)
Comment (Kuuuube) [reply ext/js/display/display-anki.js]: Switched to doing this in an array now so that's a bit cleaner.

File: ext/js/comm/anki-connect.js [hunk 3]
```diff
@@ -655,6 +666,45 @@ export class AnkiConnect {
...
                 modelName,
                 cards: cards2,
+                cardsInfo: [],
+            };
+            result2.push(item2);
+        }
+        return result2;
+    }
+
+    /**
```
Comment (jamesmaa) [inline ext/js/comm/anki-connect.js:676]: Can you add a description for what this function does? The logic is somewhat hard to grok from glimpsing

**Review by jamesmaa — APPROVED** [review]

File: ext/settings.html (hunk 2 above)
Comment (Kuuuube) [reply ext/settings.html:1963]: Changed the option.

File: ext/js/comm/anki-connect.js (hunk 3 above)
Comment (Kuuuube) [reply ext/js/comm/anki-connect.js:676]: Added.

**Review by jamesmaa — APPROVED** [review]

**Pull request merged**